From 24089750fa597c4b91e555a1177347122e863996 Mon Sep 17 00:00:00 2001
From: Michal Marek <mmarek@suse.com>
Date: Fri, 19 Feb 2016 09:30:40 +0100
Subject: [PATCH] kabi: Workaround spurious kabi change in kernel/ptrace.c
Patch-mainline: Never, kabi
References: bsc#959709, CVE-2015-8709, bsc#960561

The genksyms tool sees the definitions from <linux/user_namespace.h> and
goes crazy.

Signed-off-by: Michal Marek <mmarek@suse.com>
---
 kernel/ptrace.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/ptrace.c b/kernel/ptrace.c
index 02e965da22e9..3e1809a438ed 100644
--- a/kernel/ptrace.c
+++ b/kernel/ptrace.c
@@ -20,7 +20,9 @@
 #include <linux/uio.h>
 #include <linux/audit.h>
 #include <linux/pid_namespace.h>
+#ifndef __GENKSYMS__
 #include <linux/user_namespace.h>
+#endif
 #include <linux/syscalls.h>
 #include <linux/uaccess.h>
 #include <linux/regset.h>
-- 
2.1.4

